### ISSUE TITLE
Feature Request: Check for Duplicate Values

### ISSUE THREAD

**Issue by ciaranrh** [adrienverge/yamllint#129]
I know it's possible to check for duplicate keys but I don't see anything in the docs mentioning duplicate key values. 

For example, 
```
    http:
      paths:
      - path: /
        backend:
          serviceName: web-service1
          servicePort: 80
      - path: /
        backend:
          serviceName: web-service2
          servicePort: 80
```
It would be great if there was an option for yamllist to throw an error for the two occurrences of `- path: /` - specifically the fact the `/` occurs twice within separate `path:` keys. 

Being able to check specific keys for duplicate string patterns in their values would be a great feature as well.

**Comment by adrienverge** [comment]
Hi,

If I understand correctly you would like to detect if `path:` are the same, but don't care about `backend:`?

Do you have an idea of how you would express that in terms of configuration?

**Comment by ciaranrh** [comment]
Thanks for your reply, 

Correct, specifically the fact that `path: /` occurs twice with the same value. Multiple duplicate keys are ok, but multiple keys with the same value should throw an error.

Configuration wise, it could be a simple boolean like `key-duplicates: {}` but instead `value-duplicates: {}`.

**Comment by adrienverge** [comment]
I meant, how would you tell the conf that `path:` cannot be identical, but `backend:` can?

In YAML, this:
```yaml
- path: /
  backend: whatever1
```
is the same as:

```yaml
- backend: whatever1
  path: /
```

**Comment by ciaranrh** [comment]
My apologies, I see now. 

Perhaps in the conf, identifying the keys to verify against would be possible. As well as a flag to signal if it should be done per document in the .yaml or per the entire .yaml.

So the conf would have:
```
value-duplicates: 
  key: path
  document-level: true
```

**Comment by adrienverge** [comment]
Hi @ciaranrh, thanks for your feedback.

I understand the use-case and the need for this new option, but it seems a bit complicated to implement. We should also think about having multiple keys to check (not only `path`), nested keys, etc.

If you want to propose an implementation plan as well as a configuration spec, I will review it.

**Comment by ciaranrh** [comment]
Thanks Adrien,

Programatically, the configuration could maybe have something like:
```value-duplicates:
         keys: ["<key1 to check values","<key2 to check values>"]
         document-level: <boolean>
```
Then with the list of keys to check: 
1. Go through the .yaml document (or file depending on `document-level` boolean), verify the key exists - if so append location of .yaml to a new list
2. Go through new list and verify each value is unique compared to the other values in this list - if a value in this list occurs twice throw an error.

So item 2 is basically just a list of, the whole 'path' of the key and it's value, like `[http][paths][0][path]: /`. 

Hopefully that makes sense, that's sort of how I imagined it'd work in my head. Thanks for looking into this.

**Comment by adrienverge** [comment]
I still see potential problems with the configuration you describe, especially with complicated objects that contains `path` at several places.
For example, what `value-duplicates` conf would you write in order to check `path` values in `http.paths` and `http.paths[0].other.list`, but don't care about `http.paths[2].completely_different.a.b.c.d`?

```yaml
http:
  paths:
  - path: /
    other:
      list:
      - name: A
        path: true
      - name: B
        path: true
  - path: /
  - path: /
    completely_different:
      a:
        b:
          c:
            d:
            - path: 1
            - path: 2
```

**Comment by adrienverge** [comment]
Closing this issue due to lack of response.
